lxd/instance/drivers/driver_qemu.go

```diff
@@ -5157,3 +5157,40 @@ func (vm *qemu) writeInstanceData() error {
 
 	return nil
 }
+
+// Info returns "qemu" and the currently loaded qemu version, 0 if not found.
+func (vm *qemu) Info() instance.Info {
+	var qemuVersion, qemuBinary, qemuPath string
+	var out []byte
+	var err error
+	data := instance.Info{
+		Name:    "qemu",
+		Version: "0",
+	}
+
+	if vm.architecture == 0 {
+		vm.architecture, err = osarch.ArchitectureGetLocalID()
+	}
+	if err != nil {
+		return data
+	}
+
+	qemuBinary, _, err = vm.qemuArchConfig()
+	if err != nil {
+		return data
+	}
+
+	qemuPath, err = exec.LookPath(qemuBinary)
+	if err != nil {
+		return data
+	}
+
+	out, err = exec.Command(qemuPath, "--version").Output()
+	if err != nil {
+		return data
+	}
+
+	qemuVersion = strings.Fields(string(out))[3]
```

Comment:
This needs to be hardened. If `--version` is returning us something with less than 4 fields, this line would cause LXD to completely panic. Instead we should have `data.Version` set to `unknown` if `qemu` can be found but we can't locate the version string.

Reply:
I added a check to ensure that it won't panic, and instead returns ``data.Version = "unknown"`` if the ``--version`` output is certainly not long enough. I could try and add more checks to ensure that the field that's being taken in matches the form ``x.y.z`` and return ``unknown`` if it doesn't match, as right now it just assumes that the 4th entry in the field is the version number.

Reply:
Nah, that looks good now.